pytential/qbx/refinement.py

```diff
@@ -368,12 +368,15 @@ def check_sufficient_source_quadrature_resolution(self,
         found_panel_to_refine.finish()
 
         from pytential import bind, sym
-        dd = sym.as_dofdesc(sym.GRANULARITY_ELEMENT).to_stage2()
         source_danger_zone_radii_by_panel = flatten(
-                bind(stage2_density_discr,
-                    sym._source_danger_zone_radii(
-                        stage2_density_discr.ambient_dim, dofdesc=dd))
-                (self.array_context), self.array_context)
+                bind(
+                    stage2_density_discr,
+                    sym.ElementwiseMax(
+                        sym._source_danger_zone_radii(
+                            stage2_density_discr.ambient_dim,
+                            dofdesc=sym.QBX_SOURCE_STAGE2),
+                        dofdesc=sym.GRANULARITY_ELEMENT)
+                    )(self.array_context), self.array_context)
```

Comment:
Not sure this `source_danger_zone` needs to be `by_panel`, but changing it seems to require some work to area query template, so I left it as is for now.